src/Cli/dotnet/Program.cs

```diff
@@ -64,33 +64,6 @@ public static int Main(string[] args)
                 {
                     return ProcessArgs(args, startupTime);
                 }
-                catch (HelpException e)
```

Comment:
the use of ShowHelp() here means that to get the same semantics we needed to push these clauses to the Exception handling middleware in the Parser definition.